.github/workflows/test-emit-metrics.yml

```diff
@@ -0,0 +1,34 @@
+name: Test the emit-metrics action
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - feat-emit-metrics-action
+
+permissions:
+  id-token: write
+
+jobs:
+  emit-a-metric:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.TEST_CI_AWS_ROLE_ARN }}
+          aws-region: us-west-2
+      - name: Set up kat tool
+        uses: ./.github/actions/setup-kat
+      - name: Emit some test metrics
+        uses: ./.github/actions/emit-metrics
+        with:
+          namespace: Test Namespace (ignore me)
+          dimensions: |
+            Artifact=foo-artifact
+            Platform=Kotlin/JVM
+            Variant=External
+          metrics: |
+            BuildTime:4.532:Seconds
+            BuildSucceeded:1:Count
```

Comment:
I'm curious to see how these metrics strings will be formed in our real CI, particularly the values. Maybe using environment variables?

Reply:
I think in practice, the specific metrics names/types and dimension names will be static. The metric/dimension values will likely be dynamic based on available context. Env variables are an interesting idea. Are you suggesting we could accumulate metrics and dimensions in env vars or other context during a workflow and then emit them all in a final step without explicitly re-enumerating them?

Reply:
Yes the metric values will need to be dynamic, so I was imagining you would set an env var while executing the build-related portions of CI, then read them back in the subsequent `emit-metrics` step